src/js/tabs/onboard_logging.js

```diff
@@ -250,6 +253,85 @@ TABS.onboard_logging.initialize = function (callback) {
             loggingRatesSelect.val(BLACKBOX.blackboxRateNum + '/' + BLACKBOX.blackboxRateDenom);
         }
     }
+
+    function populateDebugModes(debugModeSelect) {
+        var debugModes = [];
+
+        if (semver.gte(CONFIG.apiVersion, "1.42.0")) {
+            $('.blackboxDebugMode').show();
+
+            debugModes = [
+                {text: "NONE"},
+                {text: "CYCLETIME"},
+                {text: "BATTERY"},
+                {text: "GYRO_FILTERED"},
+                {text: "ACCELEROMETER"},
+                {text: "PIDLOOP"},
+                {text: "GYRO_SCALED"},
+                {text: "RC_INTERPOLATION"},
+                {text: "ANGLERATE"},
+                {text: "ESC_SENSOR"},
+                {text: "SCHEDULER"},
+                {text: "STACK"},
+                {text: "ESC_SENSOR_RPM"},
+                {text: "ESC_SENSOR_TMP"},
+                {text: "ALTITUDE"},
+                {text: "FFT"},
+                {text: "FFT_TIME"},
+                {text: "FFT_FREQ"},
+                {text: "RX_FRSKY_SPI"},
+                {text: "RX_SFHSS_SPI"},
+                {text: "GYRO_RAW"},
+                {text: "DUAL_GYRO"},
+                {text: "DUAL_GYRO_RAW"},
+                {text: "DUAL_GYRO_COMBINE"},
+                {text: "DUAL_GYRO_DIFF"},
+                {text: "MAX7456_SIGNAL"},
+                {text: "MAX7456_SPICLOCK"},
+                {text: "SBUS"},
+                {text: "FPORT"},
+                {text: "RANGEFINDER"},
+                {text: "RANGEFINDER_QUALITY"},
+                {text: "LIDAR_TF"},
+                {text: "ADC_INTERNAL"},
+                {text: "RUNAWAY_TAKEOFF"},
+                {text: "SDIO"},
+                {text: "CURRENT_SENSOR"},
+                {text: "USB"},
+                {text: "SMARTAUDIO"},
+                {text: "RTH"},
+                {text: "ITERM_RELAX"},
+                {text: "ACRO_TRAINER"},
+                {text: "RC_SMOOTHING"},
+                {text: "RX_SIGNAL_LOSS"},
+                {text: "RC_SMOOTHING_RATE"},
+                {text: "ANTI_GRAVITY"},
+                {text: "DYN_LPF"},
+                {text: "RX_SPEKTRUM_SPI"},
+                {text: "DSHOT_RPM_TELEMETRY"},
+                {text: "RPM_FILTER"},
+                {text: "D_MIN"},
+                {text: "AC_CORRECTION"},
+                {text: "AC_ERROR"},
+                {text: "DUAL_GYRO_SCALED"},
+                {text: "DSHOT_RPM_ERRORS"},
+                {text: "CRSF_LINK_STATISTICS_UPLINK"},
+                {text: "CRSF_LINK_STATISTICS_PWR"},
+                {text: "CRSF_LINK_STATISTICS_DOWN"},
+                {text: "BARO"},
+                {text: "GPS_RESCUE_THROTTLE_PID"},
+                {text: "DYN_IDLE"},
+            ];
+            
```

Comment:
as noted in https://github.com/betaflight/betaflight/pull/8681#pullrequestreview-273065823, if you have a 'DEBUG_MODE_COUNT' available from the firmware you can add 'UNKNOWN' for each extra item to the `debugModes` list here.  This allows new debug modes to be added to the firmware and the configurator will still work.

Reply:
Updated with debug count